ci/functional/test_main_prep_node.sh

```diff
@@ -2,27 +2,62 @@
 
 set -eux
 
+report_junit() {
+    local msg="$1"
+    local name="$2"
+    local msg2="$3"
+
+            msg="Failed to bring up interface $iface on $HOSTNAME after it was found down.
+Please file a CORCI ticket."
+            echo -e "$msg"
+    cat <<EOF > results.xml
+  <testcase classname="$HOSTNAME" name="$name" time="0">
+    <failure message="$msg2" type="TestFail"><![CDATA[$msg]]></failure>
+  </testcase>
+EOF
+}
+
 for i in 0 1; do
     iface="ib$i"
     if [ -e /sys/class/net/"$iface" ]; then
         if ! ifconfig "$iface" | grep "inet "; then
-          {
-            echo "Found interface $iface down after reboot on $HOSTNAME"
-            systemctl status || true
-            systemctl --failed || true
-            journalctl -n 500 || true
-            ifconfig "$iface" || true
-            cat /sys/class/net/"$iface"/mode || true
-            ifup "$iface" || true
-            ifconfig -a || true
-            ls -l /etc/sysconfig/network-scripts/ifcfg-* || true
-            cat /etc/sysconfig/network-scripts/ifcfg-"$iface" || true
-          } 2>&1 | mail -s "Interface $iface found down after reboot" \
-                   -r "$HOSTNAME"@intel.com "$OPERATIONS_EMAIL"
+            mkdir -p /tmp/artifacts
+            systemctl status > /tmp/artifacts/systemctl\ status || true
+            systemctl --failed > /tmp/artifacts/systemctl\ --failed || true
+            journalctl -n 500 > /tmp/artifacts/journalctl\ -n\ 500 || true
+            dmesg > /tmp/artifacts/dmesg || true
+            ifconfig "$iface" > /tmp/artifacts/ifconfig\ "$iface" || true
+            cat /sys/class/net/"$iface"/mode > \
+                /tmp/artifacts/cat\ _sys_class_net_"${iface}"_mode || true
+            ifup "$iface" > /tmp/artifacts/ifup\ "$iface" || true
+            ifconfig -a > /tmp/artifacts/ifconfig\ -a || true
+            ls -l /etc/sysconfig/network-scripts/ifcfg-* > \
+                /tmp/artifacts/ls\ -l\ _etc_sysconfig_network-scripts_ifcfg-\* || true
+            cat /etc/sysconfig/network-scripts/ifcfg-"$iface" > \
+                /tmp/artifacts/cat\ _etc_sysconfig_network-scripts_ifcfg-"$iface" || true
+            send_mail "Interface $iface found down after reboot"  \
+                      "Found interface $iface down after reboot."
         fi
         if ! ifconfig "$iface" | grep "inet "; then
-            echo "Failed to bring up interface $iface on $HOSTNAME. " \
-            "Please file a CORCI ticket."
+            report_junit "Failed to bring up interface $iface on $HOSTNAME after it was found down. 
+Please file a CORCI ticket." "IB_interface_down" "Interface $iface on $HOSTNAME is down and won't come up. 
```

Comment:
(style)  trailing whitespace
(style)  line over 100 characters

Reply:
Intentional.  It's inside a multiline string.